server/src/main/java/org/opensearch/search/profile/AbstractProfileBreakdown.java

```diff
@@ -69,13 +69,27 @@ public void setTimer(T timing, Timer timer) {
     }
 
     /**
-     * Build a timing count breakdown.
+     * Return a timing count breakdown.
      */
     public final Map<String, Long> toBreakdownMap() {
-        Map<String, Long> map = new HashMap<>(timings.length * 2);
-        for (T timingType : timingTypes) {
-            map.put(timingType.toString(), timings[timingType.ordinal()].getApproximateTiming());
-            map.put(timingType.toString() + "_count", timings[timingType.ordinal()].getCount());
+        return doBuildBreakdownMap();
+    }
+
+    /**
+     * Build a timing count breakdown for current instance
+     */
+    protected Map<String, Long> doBuildBreakdownMap() {
```

Comment:
What is the purpose of introducing this method as opposed to making `toBreakdownMap` non-final and overriding it in ConcurrentQueryProfileBreakdown?

Reply:
Good one, only to not break the API compatibility, this is the only reason, thank you

Reply:
What would break by making `toBreakdownMap` non-final? If there was ever any intent to making that method final, it is being circumvented by this new protected method, so I'd be inclined to simplify things and just make it non-final. Unless there is some compatibility concern that I don't understand.

Reply:
No objections, will do that, thanks!

Reply:
Fixed!